### DISCUSSION TITLE
background task blocking requests , when GZipMiddleware and another middleware is added

### DISCUSSION THREAD

**Discussion by vedil** [fastapi/fastapi#10144] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import asyncio
import fastapi  # fastapi==0.75.2
import time
import uvicorn  # uvicorn==0.17.6
from starlette.middleware.gzip import GZipMiddleware
from app.RequestContextMiddleware import RequestContextMiddleware
import logging
from fastapi import (
    APIRouter,
    BackgroundTasks,
    Depends,
    File,
    Request,
    Response,
    UploadFile,
)

logger = logging.getLogger("uvicorn")
logger.setLevel(logging.DEBUG)
logger.handlers.clear()

log_formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
handler = logging.StreamHandler()
handler.setFormatter(log_formatter)
logger.addHandler(handler)
logging.getLogger("multipart").setLevel(logging.WARN)

app = fastapi.FastAPI()

#app.add_middleware(RequestContextMiddleware)
app.add_middleware(GZipMiddleware, minimum_size=1000)

@app.middleware("http")
async def add_process_time_header(request: Request, call_next):
    start_time = time.time()
    response = await call_next(request)
    process_time = time.time() - start_time
    response.headers["X-Process-Time"] = str(process_time)
    return response

async def sample_task(sleep_time: int ):
    logger.debug("in task start of b")
    await asyncio.sleep(sleep_time)
    logger.debug("in task end of b")

@app.get("/{sleep_time}")
async def page_test(
    background_tasks: BackgroundTasks,
    sleep_time:int,
):
    logger.debug("start of b")
    background_tasks.add_task( sample_task, sleep_time)
    logger.debug("end of b")
    return {"status":"OK"}
```

### Description

similar to https://github.com/encode/starlette/issues/919

using curl we will not see this behavior (curl -v http://localhost:7016/10)

but using postman or swagger UI will block the response (response header comes but body is only gotten after sleep_time)

some times first call go through fine, but successive ones fail

### Operating System

macOS

### FastAPI Version

0.101.0

### Pydantic Version

1.10.2

### Python Version

Python 3.11.2

### Additional Context

[as ](https://github.com/encode/starlette/issues/919#issuecomment-665477901)

**Comment by Kludex** [comment, marked as answer]
This is solved in the latest Starlette. When FastAPI bumps it, it will be alright.
